rundeckapp/grails-app/services/rundeck/services/FrameworkService.groovy

```diff
@@ -775,6 +765,25 @@ class FrameworkService implements ApplicationContextAware {
     def Description getStepPluginDescription(String type) throws MissingProviderException{
         rundeckFramework.getStepExecutionService().providerOfType(type).description
     }
+
+    /**
+     * Return step plugin of a certain type
+     * @param type
+     * @return
+     */
+    def getStepPlugin(String type) throws MissingProviderException{
+        rundeckFramework.getStepExecutionService().providerOfType(type)
+    }
+
+    /**
+     * Return dynamic properties values
+     * @param type, projectAndFrameworkValues
+     * @return
+     */
+    def Map<String, Object> getDynamicProperties(String type, Map<String, Object> projectAndFrameworkValues){
+        def plugin = getStepPlugin(type)
+        plugin.getDynamicProperties(projectAndFrameworkValues)
```

Comment:
if the plugin does not implement the interface, this will cause a runtime error

Reply:
But this object is a StepPluginAdapter type and on class StepPluginAdapter.java line 66 this condition is tested. This could be of other type?

Reply:
yes, there are a few other `NodeStepExecutor`s that could be returned

Reply:
Another issue: this dynamic properties feature should also support "NodeStep" step types .e.g. see `NodeStepPluginAdapter`